File: app/src/main/java/com/example/ase_project_nutrismart/Response/BmiData.java
package com.example.ase_project_nutrismart.Response;


import androidx.room.ColumnInfo;
import androidx.room.Entity;

import java.io.Serializable;

@Entity
public class BmiData implements Serializable {

    public double getbMI() {
        return bMI;
    }

    public void setbMI(double bMI) {
        this.bMI = bMI;
    }

    public double getProteinNeeded() {
        return proteinNeeded;
    }

    public void setProteinNeeded(double proteinNeeded) {
        this.proteinNeeded = proteinNeeded;
    }

    public double getCarbsNeeded() {
        return carbsNeeded;
    }

    public void setCarbsNeeded(double carbsNeeded) {
        this.carbsNeeded = carbsNeeded;
    }

    public double getFatsNeeded() {
        return fatsNeeded;
    }

    public void setFatsNeeded(double fatsNeeded) {
        this.fatsNeeded = fatsNeeded;
    }

    public double getCalorie() {
        return calorie;
    }

    public void setCalorie(double calorie) {
        this.calorie = calorie;
    }

    @ColumnInfo(name="bMI")
    public double bMI;

    @ColumnInfo(name="proteinNeeded")
    public double proteinNeeded;

    @ColumnInfo(name="carbsNeeded")
    public double carbsNeeded;

    @ColumnInfo(name="fatsNeeded")
    public double fatsNeeded;


    @ColumnInfo(name="calorie")
    public double calorie;
}

File: app/src/main/java/com/example/ase_project_nutrismart/Response/BmiData.kt
package com.example.ase_project_nutrismart.Response

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.example.ase_project_nutrismart.Response.BmiData
import com.example.ase_project_nutrismart.Response.ExpireData
import com.example.ase_project_nutrismart.Response.Grocery
import com.example.ase_project_nutrismart.Response.RolesResponse
import java.io.Serializable

@Entity
class BmiData : Serializable {
    fun getbMI(): Double {
        return bMI
    }

    fun setbMI(bMI: Double) {
        this.bMI = bMI
    }

    @JvmField
    @ColumnInfo(name = "bMI")
    var bMI = 0.0

    @JvmField
    @ColumnInfo(name = "proteinNeeded")
    var proteinNeeded = 0.0

    @JvmField
    @ColumnInfo(name = "carbsNeeded")
    var carbsNeeded = 0.0

    @JvmField
    @ColumnInfo(name = "fatsNeeded")
    var fatsNeeded = 0.0

    @JvmField
    @ColumnInfo(name = "calorie")
    var calorie = 0.0
}